File: DesignPattenStudy/AdapterPattern/Java/AdapterImpl.java
package AdapterPattern.Java;

public class AdapterImpl implements Adapter {

    @Override
    public Float twiceOf(Float f) {

        return (float) Math.twoTime(f.doubleValue());
    }

    @Override
    public Float halfOf(Float f) {

        return (float) Math.halfTime(f.doubleValue());
    }
}

File: DesignPattenStudy/AdapterPattern/Kotlin/AdapterImpl.kt
package AdapterPattern.Kotlin

class AdapterImpl : Adapter {

    override fun twiceOf(f: Float): Float {

        return Math.twiceOf(f.toDouble()).toFloat()
    }

    override fun halfOf(f: Float): Float {

        return Math.halfOf(f.toDouble()).toFloat()
    }
}